pylint/checkers/typecheck.py

```diff
@@ -523,6 +525,79 @@ def _emit_no_member(
     return True
 
 
+def _get_all_attribute_assignments(
+    node: nodes.FunctionDef, name: str | None = None
+) -> set[str]:
+    attributes: set[str] = set()
+    for child in node.nodes_of_class((nodes.Assign, nodes.AnnAssign)):
+        targets = []
+        if isinstance(child, nodes.Assign):
+            targets = child.targets
+        elif isinstance(child, nodes.AnnAssign):
+            targets = [child.target]
+        for assign_target in targets:
+            if isinstance(assign_target, nodes.Tuple):
+                targets.extend(assign_target.elts)
+                continue
+            if (
+                isinstance(assign_target, nodes.AssignAttr)
+                and isinstance(assign_target.expr, nodes.Name)
+                and (name is None or assign_target.expr.name == name)
+            ):
+                attributes.add(assign_target.attrname)
+    return attributes
+
+
+def _enum_has_attribute(
+    owner: astroid.Instance | nodes.ClassDef, node: nodes.Attribute
+) -> bool:
+    if isinstance(owner, astroid.Instance):
+        enum_def = next(
+            (b.parent for b in owner.bases if isinstance(b.parent, nodes.ClassDef)),
+            None,
+        )
+
+        if enum_def is None:
+            # We don't inherit from anything, so try to find the parent
+            # class definition and roll with that
+            enum_def = node
+            while enum_def is not None and not isinstance(enum_def, astroid.ClassDef):
```

Comment:
```suggestion
            while not isinstance(enum_def, nodes.ClassDef):
```

Reply:
This should also work right? And I added `nodes.`.

Reply:
From a correctness PoV, we need the `not None` check because if there is no classdef and we hit the module node, then `.parent` will be `None` and we'll crash out with a "`NoneType` doesn't have attribute parent" exception. However in reality that "shouldn't happen™" because a class has to have a definition node somewhere.

And thanks, I need to programme my brain to use `nodes` instead of `astroid` 😬

Reply:
Oh of course.. I mixed up my `not None` and `is not`'s... We should probably keep this in then 😅